cura/Settings/CustomSettingFunctions.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+from typing import Any, List, Optional, TYPE_CHECKING
+
+from UM.Settings.PropertyEvaluationContext import PropertyEvaluationContext
+from UM.Settings.SettingFunction import SettingFunction
+
+if TYPE_CHECKING:
+    from cura.CuraApplication import CuraApplication
+    from cura.Settings.CuraContainerStack import CuraContainerStack
+
+
+#
+# This class contains all Cura-related custom setting functions. Some functions requires information such as the
```

Comment:
Setting Function should be together (Since you are refering to UM.Settings.SettingFunction)

Reply:
That's what a SettingFunction is. At least, that's what the documentation of the class says; "Encapsulates Python code that provides a simple value calculation function."

Reply:
That's not correct. If a setting's property is a function, that field will be evaluated using `SettingFunction`. So, you can say that `SettingFunction` represents the value of the a setting property which is a Python code that needs to be evaluated. It does not represent any single (custom) function in a formula.

I'll update the documentation for this. As for `CustomSettingFunctions`, maybe `CustomFormulaFunctions` can be a more accurate name?

Reply:
`SettingFunction` doc updated in https://github.com/Ultimaker/Uranium/commit/be7934eff93b50466c4dca8b3c165ed097f32372